include/circt/Dialect/FIRRTL/FIRRTLTypes.h

```diff
@@ -88,6 +94,9 @@ class FIRRTLType : public Type {
     return getRecursiveTypeProperties().containsReference;
   }
 
+  /// Return true if this is an anonymous type (no type alias).
+  bool isAnonymousType() { return getRecursiveTypeProperties().isAnonymous; }
```

Comment:
Does this only apply to the outer part of a type?  "type contains a type alias" implies a field of a bundle could be a type alias and this would be... false?

Why isn't this named so that it refers to specific terms used in the hierarchy?  containsAlias? isAlias?

 If this is "isAlias", that is just a isa check on a type, so it doesn't need to be a property.  So I assume you mean containsAlias.

Reply:
FWIW, I agree and this was briefly discussed above (if I'm understanding you): https://github.com/llvm/circt/pull/5167/files#r1190024453

Reply:
Although good point re:if it's just isa then this isn't a recursive type property.  :thinking: .